src/app/modules/item/pages/item-progress/item-progress.component.ts

```diff
@@ -1,18 +1,53 @@
-import { Component, Input } from '@angular/core';
-import { delay } from 'rxjs/operators';
+import { Component, Input, OnChanges, ViewChild } from '@angular/core';
+import { delay, map, switchMap, filter } from 'rxjs/operators';
 import { UserSessionService } from 'src/app/shared/services/user-session.service';
 import { ItemData } from '../../services/item-datasource.service';
+import { RouterLinkActive } from '@angular/router';
+import { Observable, ReplaySubject } from 'rxjs';
+import { ItemType } from '../../../../shared/helpers/item-type';
 
 @Component({
   selector: 'alg-item-progress',
   templateUrl: './item-progress.component.html',
   styleUrls: [ './item-progress.component.scss' ]
 })
-export class ItemProgressComponent {
+export class ItemProgressComponent implements OnChanges {
 
   @Input() itemData?: ItemData;
 
-  session$ = this.sessionService.session$.pipe(delay(0));
+  @ViewChild('historyTab') historyTab?: RouterLinkActive;
+  @ViewChild('chapterGroupProgressTab') chapterGroupProgressTab?: RouterLinkActive;
+  @ViewChild('chapterUserProgressTab') chapterUserProgressTab?: RouterLinkActive;
 
+  readonly session$ = this.sessionService.session$.pipe(delay(0));
+  private readonly type$ = new ReplaySubject<ItemType>(1);
+  readonly selectors$ = this.getSelectors$();
+  
   constructor(private sessionService: UserSessionService) {}
+
+  ngOnChanges(): void {
+    if (!this.itemData) {
+      return;
+    }
+
+    this.type$.next(this.itemData.item.type);
+  }
+
+  private getSelectors$(): Observable<'none' | 'withUserProgress' | 'withGroupProgress'> {
+    return this.type$.pipe(
+      filter(type => !!type),
+      switchMap(type => this.sessionService.watchedGroup$.pipe(
+        map(watchedGroup => {
+          if (!watchedGroup && [ 'Task', 'Course' ].includes(type)) {
+            return 'none';
+          } else if (!watchedGroup && type === 'Chapter') {
+            return 'withUserProgress';
+          } else {
+            return 'withGroupProgress';
+          }
+        })
+      )),
+      delay(0)
+    )
```

Comment:
Except it I missed things, there is no need for subjects and observable here (it overcomplexifies things), on init and on change just call a "recomputeSelector" function which will take this.itemData.item.type and call a function (to be created) `isCurrentWatching` on sessionService. `isCurrentWatching` just use current value of its subject to respond.

Reply:
Fixed